src/main/java/org/minions/devfund/roy/WhackAMole.java

```diff
@@ -0,0 +1,169 @@
+package org.minions.devfund.roy;
+
+import java.util.Random;
+
+/**
+ * WhackAMole game.
+ */
+public class WhackAMole {
+
+    private int score;
+    private int molesLeft;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+    private static final char MOLE = 'M';
+    private static final char WHACKED_MOLE = 'W';
+    private static final char EMPTY_PLACE = '*';
+
+    /**
+     * Constructor.
+     *
+     * @param numAttempts   to start the game.
+     * @param gridDimension to create the moleGrid for the game.
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        this.moleGrid = new char[gridDimension][gridDimension];
+        attemptsLeft = numAttempts;
+        this.molesLeft = 0;
+        initializeGrid();
+    }
+
+    /**
+     * Verifies if a place is empty or not.
+     *
+     * @param x axis position.
+     * @param y axis position.
+     * @return true if the place is empty otherwise false.
+     */
+    public boolean place(int x, int y) {
+        if (x >= 0 && y >= 0 && x < moleGrid.length && y < moleGrid.length
+                && moleGrid[x][y] == EMPTY_PLACE) {
+            moleGrid[x][y] = MOLE;
+            molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Whacks a received position.
+     *
+     * @param x axis position.
+     * @param y axis position.
+     */
+    public void whack(int x, int y) {
+        if (rules() && !giveUp(x, y)) {
+            if (moleGrid[x][y] == MOLE) {
+                moleGrid[x][y] = WHACKED_MOLE;
+                score++;
+                attemptsLeft--;
+                molesLeft--;
+            } else {
```

Comment:
this else conditional can be removed
you can declare directly attemptsLeft--;
and remove it from if conditional.